pkg/sitewise/datasource.go

```diff
@@ -20,21 +23,86 @@ type invokerFunc func(ctx context.Context, sw client.SitewiseClient) (framer.Fra
 
 type Datasource struct {
 	GetClient clientGetterFunc
+	closeCh   chan struct{}
+	closeOnce sync.Once
 }
 
 func NewDatasource(settings backend.DataSourceInstanceSettings) (*Datasource, error) {
-	cfg := client.AWSSiteWiseDataSourceSetting{}
+	cfg := models.AWSSiteWiseDataSourceSetting{}
+
 	err := cfg.Load(settings)
 	if err != nil {
 		return nil, err
 	}
+
+	err = cfg.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	done := make(chan struct{})
 	sessions := awsds.NewSessionCache()
+	clientGetter := func(region string) (swclient client.SitewiseClient, err error) {
+		swclient, err = client.GetClient(region, cfg, sessions.GetSession)
+		return
+	}
 
-	return &Datasource{
-		GetClient: func(region string) (swclient client.SitewiseClient, err error) {
-			swclient, err = client.GetClient(region, cfg, sessions.GetSession)
+	if cfg.Region == models.EDGE_REGION && cfg.EdgeAuthMode != models.EDGE_AUTH_MODE_DEFAULT {
+		edgeAuthenticator := dummyAuthenticator{ //EdgeAuthenticator{
```

Comment:
test this manually with the dummyAuthenticator, and lets then commit with EdgeAuthenticator

In another PR, lets explore how to do this on demand rather than as a background task.